Source/WebGPU/WGSL/WGSL.h

```diff
@@ -145,8 +145,11 @@ struct BindGroupLayoutEntry {
     OptionSet<ShaderStage> visibility;
     using BindingMember = std::variant<BufferBindingLayout, SamplerBindingLayout, TextureBindingLayout, StorageTextureBindingLayout, ExternalTextureBindingLayout>;
     BindingMember bindingMember;
+    std::optional<uint32_t> vertexArgumentBufferIndex;
     std::optional<uint32_t> vertexBufferDynamicOffset;
+    std::optional<uint32_t> fragmentArgumentBufferIndex;
     std::optional<uint32_t> fragmentBufferDynamicOffset;
+    std::optional<uint32_t> computeArgumentBufferIndex;
     std::optional<uint32_t> computeBufferDynamicOffset;
```

Comment:
Are these all individually optional or are groups of them related? (see **) 
The naming pattern hints that this really an array indexed by ShaderStage.

Reply:
They are individually optional, but I could store them in an array instead. I'm not sure what is easier for the compiler?